test/MinimalDelegation.execute.t.sol

```diff
@@ -517,4 +517,105 @@ contract MinimalDelegationExecuteTest is TokenHandler, DelegationHandler, Execut
         signerAccount.execute(BATCHED_CALL_SUPPORTS_OPDATA, executionData);
         vm.snapshotGasLastCall("execute_BATCHED_CALL_SUPPORTS_OPDATA_twoCalls");
     }
+
+    function test_execute_batchOfBatches_succeeds() public {
+        Call[] memory calls = CallBuilder.init();
+        calls = calls.push(buildTransferCall(address(tokenA), address(receiver), 1e18)); // Transfer 1 tokenA
+
+        // Get the current nonce components for key 0
+        uint256 nonceKey = 0;
+        (uint256 nonce,) = _buildNextValidNonce(nonceKey);
+
+        // Create hash of the calls + nonce and sign it
+        SignedCalls memory signedCalls = SignedCalls({calls: calls, nonce: nonce});
+        bytes32 hashToSign = signerAccount.hashTypedData(signedCalls.hash());
+        bytes memory signature = signerTestKey.sign(hashToSign);
+        // Pack the execution data:
+        // 1. Encode the nonce and signature into opData
+        bytes memory opData = abi.encode(nonce, signature);
+        // 2. Encode the calls and opData together
+        bytes memory executionData1 = abi.encode(calls, opData);
+
+        calls = CallBuilder.init();
+        calls = calls.push(buildTransferCall(address(tokenA), address(receiver), 2e18)); // Transfer 2 tokenA
+
+        // Create hash of the calls + nonce and sign it
+        nonce++; // increment the nonce since first one hasnt been used yet
+        signedCalls = SignedCalls({calls: calls, nonce: nonce});
+        hashToSign = signerAccount.hashTypedData(signedCalls.hash());
+        signature = signerTestKey.sign(hashToSign);
+
+        // Pack the execution data:
+        // 1. Encode the nonce and signature into opData
+        opData = abi.encode(nonce, signature);
```

Comment:
same need to wrap the signature

Reply:
also for the second batch would be sweet if this tested on a different key than the first call